### PR TITLE
Fix qt_assert when opening preferences #11454

### PR DESCRIPTION
**Pull request by daschuer**
This fixes https://github.com/mixxxdj/mixxx/issues/11454

### CODE REVIEW

File: src/preferences/effectmanifesttablemodel.cpp [hunk 1]
```diff
@@ -146,8 +146,17 @@ Qt::DropActions EffectManifestTableModel::supportedDropActions() const {
 }
 
 bool EffectManifestTableModel::removeRows(int row, int count, const QModelIndex& parent) {
-    Q_UNUSED(parent);
-    beginRemoveRows(QModelIndex(), row, row + count - 1);
+    if (!count) {
```
Comment (Swiftb0y) [inline src/preferences/effectmanifesttablemodel.cpp:149]: implicit conversion to `int` is often a bug and is usually forbidden with `-Wconversion`. Please be explicit. 
```suggestion
    if (count == 0) {
```

File: src/preferences/effectmanifesttablemodel.cpp [hunk 2]
```diff
@@ -146,8 +146,17 @@ Qt::DropActions EffectManifestTableModel::supportedDropActions() const {
...
+        // nothing to do
+        return true;
+    }
+    VERIFY_OR_DEBUG_ASSERT(count > 0 && row < m_manifests.size() &&
+            row + count <= m_manifests.size()) {
```
Comment (Swiftb0y) [inline src/preferences/effectmanifesttablemodel.cpp:154]: The second condition is already being enforced by the result of the first and the third one. I'd rather check we're not being passed some negative row index.
```suggestion
    VERIFY_OR_DEBUG_ASSERT(count > 0 && row >= 0 &&
            row + count <= m_manifests.size()) {
```

File: src/preferences/effectmanifesttablemodel.cpp [hunk 3]
```diff
@@ -28,7 +28,7 @@ void EffectManifestTableModel::setList(const QList<EffectManifestPointer>& newLi
 
 int EffectManifestTableModel::rowCount(const QModelIndex& parent) const {
     Q_UNUSED(parent);
-    return m_manifests.count();
+    return m_manifests.size();
```
Comment (Swiftb0y) [inline src/preferences/effectmanifesttablemodel.cpp:31]: why this noop change? IMO the method is already called `rowCount`.

**Review by Swiftb0y — COMMENTED** [review]
nitpicks. LGTM otherwise.

**Comment by daschuer** [comment]
Done.

**Review by Swiftb0y — APPROVED** [review]
Thank you. Would you @JoergAtGithub be able to verify whether this is fixing the issue?

**Comment by daschuer** [comment]
Friendly ping @JoergAtGithub

**Comment by Swiftb0y** [comment]
Seems like Jörg is busy. Lets just merge regardless.

**Pull request merged**